ui/src/app/workflows/components/submit-workflow-panel.tsx

```diff
@@ -28,13 +31,27 @@ const defaultTemplate: Template = {
 
 export function SubmitWorkflowPanel(props: Props) {
     const {navigation} = useContext(Context);
-    const [entrypoint, setEntrypoint] = useState(workflowEntrypoint);
+    const [entrypoint, setEntrypoint] = useState(props.entrypoint || workflowEntrypoint);
     const [parameters, setParameters] = useState<Parameter[]>([]);
     const [workflowParameters, setWorkflowParameters] = useState<Parameter[]>(JSON.parse(JSON.stringify(props.workflowParameters)));
     const [labels, setLabels] = useState(['submit-from-ui=true']);
     const [error, setError] = useState<Error>();
     const [isSubmitting, setIsSubmitting] = useState(false);
 
+    useEffect(() => {
+        const templatePropertiesInQuery = getWorkflowParametersFromQuery(props.history);
+        // Get the user arguments from the query params
+        const updatedParams = workflowParameters.map(param => {
+            const queryValue = templatePropertiesInQuery[param.name];
+            const p: Parameter = {
+                name: param.name,
+                value: queryValue || param.value
+            };
+            return p;
+        });
```

Comment:
You can simplify this a bit by getting rid of the explicit type, because TypeScript is smart enough to know this matches `Parameter[]`:
```suggestion
        const updatedParams = workflowParameters.map(param => ({
            name: param.name,
            value: templatePropertiesInQuery[param.name] || param.value
        }));
```